### PR TITLE
Add test coverage for consuming email with attachments

### PR DESCRIPTION
**Pull request by jamesnetherton**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: integration-tests/mail/src/main/java/org/apache/camel/quarkus/component/mail/MockMailbox.java [hunk 1]
```diff
@@ -16,40 +16,90 @@
...
 import org.jvnet.mock_javamail.Mailbox;
 
-@Path("/mock/{username}")
+@Path("/mock")
 @ApplicationScoped
 public class MockMailbox {
 
+    @DELETE
+    @Path("/clear")
+    @Produces(MediaType.TEXT_PLAIN)
```
Comment (aldettinger) [inline integration-tests/mail/src/main/java/org/apache/camel/quarkus/component/mail/MockMailbox.java:48]: It is possible to @Produces with a void method ?

Comment (jamesnetherton) [reply integration-tests/mail/src/main/java/org/apache/camel/quarkus/component/mail/MockMailbox.java:48]: Nope - too much copy / paste. I'll remove it.

**Comment by jamesnetherton** [comment]
> I wonder if there would be interest in testing with a secure flavor, I mean imaps, pop3s and on.

Yes, in general the extension needs more test coverage to cover functionality for each of the supported mail protocols.

**Comment by jamesnetherton** [comment]
Looks like there's an NPE to resolve (which is not occurring for me locally)....

**Comment by jamesnetherton** [comment]
I have reworked the tests to switch from `mock-javamail` to using a containerized mock mail server. Hopefully this helps to fix things. It also opens the door to testing with additional protocols and with security enabled etc.

**Comment by jamesnetherton** [comment]
Actually after doing 50 test runs there still seems to be some instability. Not sure if it's the way I have composed the tests or some issue in `camel-mail`. I'm going to abandon this for now....

**Pull request closed without merging**